GameEngine/inc/StateMachine.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef STATEMACHINE_HPP_
+#define STATEMACHINE_HPP_
+
+#include <iostream>
+#include <memory>
+#include <stack>
+#include <stdexcept>
+
+#include "State.hpp"
+
+class StateMachine {
+ public:
+  StateMachine();
+  const bool running() { return m_running; }
+  void run(std::unique_ptr<State> state);
+  void nextState();
+  void update();
+  void draw();
+  void quit();
+  void lastState();
+  template<typename T>
```

Comment:
please add an empty line above so its easier to see that all 3 lines are one method